### ISSUE TITLE
instantOS hangs when booting with Nvidia drivers

### ISSUE THREAD

**Issue by robcsi** [instantOS/instantOS#65]
**The OS hangs after installation when using Nvidia drivers (both proprietary and non); boots ok with Nouveau drivers**
I've installed the OS on my Thinkpad T410 with a Nvidia GT218M (NVS 3100M) card.
When booting after installation it hangs with both Nvidia drivers, but not with Nouveau. There is a different problem with Nouveau, though: it performs very slowly out of the box, like performance dropping when mouse slides over a link on a browser page, plymouth anymation is slow, or some more GPU-intensive operation is performed; it kind of has these shorter hangs, glitches...

**To Reproduce**
1. Install OS on similar setup and just use it out of the box
2. Try all drivers provided with installer

**Expected behavior**
When booting the live ISO on the same machine it performs well, although I didn't check which driver is loaded. So, I would expect the same performance.

**Screenshots**
N/A.

**Device (only relevant in some cases):**
 - GPU: Nvidia GT218M (NVS 3100M)
 - CPU:  Intel(R) Core(TM) i5 CPU       M 520  @ 2.40GHz
 - Thinkpad T410, 8 Gb RAM, SSD

**Additional context**
The solution is to install with Nouveau drivers (or without any graphics driver if you feel up to it), and install the latest and greatest Nvidia driver from their donwloads page.
 - download your card's drivers
 - stop lighdm service: systemctl stop lightdm-plymouth
 - switch to another terminal: Ctrl+F2, or whichever works for you, of just not start X
 - install driver
 - reboot

I don't consider this a bug in the OS really, just posting the info for others. This has happened to me a lot of times with many distros.

**Comment by paperbenni** [comment]
The iso actually uses the open source drivers, that's why it worked. 
So you installed the driver independent of package management right?
Which driver worked? Does this card require a different driver than the usual gtx cards?
Maybe I can add this as a choice to the installer

**Comment by robcsi** [comment]
Yes, I went to official Nvidia page for driver downloads and searched for my graphics card, mentioned above. I downloaded the .run file, NVIDIA-Linux-x86_64-340.108.run in my case and installed that.

Yesterday I had another problem: I updated the system (yay command) and a new kernel was installed. After reboot Lightdm failed to load.
So, I had a hunch and reinstalled the Nvidia driver mentioned above, which made it work again.
It seems that the installed driver wasn't included/started as a module by the newly installed kernel.

How can this be avoided every time a new kernel is released?

**Comment by paperbenni** [comment]
https://download.nvidia.com/XFree86/Linux-x86_64/415.13/README/installdriver.html
You might look into the dkms

**Comment by paperbenni** [comment]
So in your case that would be installing dkms and reinstalling the nvidia driver and then it should persist between kernel updates

**Comment by robcsi** [comment]
For some reason it cannot install the driver with dkms: kernel preparation unnecessary for this kernel, error message.

Some say on forums that the kernel is too new for this driver, or rather vice-versa. :)

I'll dig some more...

**Comment by cid0rz** [comment]
I was not able to boot also with Nvidia proprietary drivers for a  NVIDIA GeForce 210. Hardware is old ( Pentium E5700 (2) @ 2.885GHz) but I was expecting it to be less resource intensive in general. I managed to install with nouveau and boot but I have 4 Gb of ram and 3.15 are in use.

**Comment by paperbenni** [comment]
Have you tried @robcsi 's method of installing the correct drivers from the nvndia website (using the blob)?

**Comment by cid0rz** [comment]
Hi, I have not tried yet. I will and report back, this is an older computer of mine so I work on it when i have time. For sure something is not right since it looks is struggling to run with only internet browser opened.

**Comment by robcsi** [comment]
@cid0rz, the lagginess may be cause by the lack of proper drivers, I also had that with the Nouveau driver.

Search for you GPU model on NVidia site and try installing that.

Note that you may need to disable lightdm-plymouth service to be able to install it.

**Comment by cid0rz** [comment]
Hello all and thank you for your replies. I tried installing but dkms was in use so i rebooted into console mode (runlevel 3) as shown [here](https://superuser.com/questions/974797/how-to-boot-a-linux-system-without-graphical-server) and managed to install the nvidia driver properly. When i boot I see a huge performance boost. It is much much better but now I see the RAM is starting to get busy without anything running but this firefox window. It is very strange, the used memory constantly increases so it looks like a memory leak of some kind. What can i do to get more information for you?

**Comment by cid0rz** [comment]
Hello back, I rebooted today and found out i have some problem with insntantwallpaper which is what is cluttering the ram, should I open a separate issue?

**Comment by paperbenni** [comment]
It's already fixed in master, try installing instantOS dev tools (settings->instantos->dev tools) and then run
```sh
ibuild install instantwallpaper
```

**Comment by cid0rz** [comment]
It worked very well apparently, thank you! I will keep testing and report back if i see anything strange.

**Comment by cid0rz** [comment]
As a recap of what I did to fix the error with drivers ( i consider instantwallpaper bug like separate):

* go to Nvidia and [search](https://www.nvidia.com/Download/index.aspx?lang=en-us) for your driver, download the .run driver for your card, in my case NVIDIA-Linux-x86_64-340.108.run
*  restart your computer and while grub splash screen press `e` on keyboard and add a 3 to the `linux` line as explained [here](https://superuser.com/questions/974797/how-to-boot-a-linux-system-without-graphical-server).
* `cd` to the downloads folder or wherever you got the .run and `sh NVIDIA-youdriver.run`
follow the instructions and install with dkms support. when it is finished installing you can `sudo reboot now` and your system should boot with the drivers.
* open a terminal and check your drivers with the command `nvidia-smi`

```
Mon Oct 26 14:03:43 2020
+------------------------------------------------------+
| NVIDIA-SMI 340.108    Driver Version: 340.108        |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce 210         Off  | 0000:01:00.0     N/A |                  N/A |
|  1%   53C   P12    N/A /  N/A |    129MiB /   511MiB |     N/A      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Compute processes:                                               GPU Memory |
|  GPU       PID  Process name                                     Usage      |
|=============================================================================|
|    0            Not Supported                                               |
+-----------------------------------------------------------------------------+
```
And now you should see a huge improvement in performance.

**Comment by robcsi** [comment]
The dmks approach didn't work for me, as I wrote above, so I have to reinstall every time the kernel is updated.

**Comment by robcsi** [comment]
I managed to install another package instead of the one from the NVidia homepage.

The AUR contains packages for this version, for both 'linux' and 'linux-lts' kernels with dkms.
I used https://aur.archlinux.org/packages/nvidia-340xx-dkms/, sticking to 'linux' kernel, to see what happens next time the kernel is updated.

I'll be back with feedback on that.

**Comment by robcsi** [comment]
Everything works fine when the kernel is updated, no need to reinstall the driver again, manually.

**Comment by Sarthak2319** [comment]
Just in case anyone is wondering, the performance issues are happening because robsci has a GPU which is not supported by the latest Nvidia driver, therefore installing the older one fixed the issue.
Disabling mode setting might also be required as Plymouth doesn't work well without mode setting and If I remember correctly mode setting support landed in Nvidia driver 390(Nvidia had already dropped support for his GPU by then).
